packages/rum-react/test/e2e/with-router/switch.e2e-spec.js

```diff
@@ -59,4 +58,32 @@ describe('Using Switch component of react router', function() {
     expect(routeTransaction.spans.length).toBe(1)
     expect(routeTransaction.spans[0].name).toBe('GET /test/e2e/data.json')
   })
+
+  it('should capture spans inside lazy rendered components', function() {
+    browser.url('/test/e2e/with-router/switch.html')
+    browser.waitUntil(
+      () => {
+        $('#lazy-func').click()
+        const componentContainer = $('#func-container')
+        return componentContainer.getText().indexOf('/lazy') !== -1
+      },
+      5000,
+      'expected functional component to be rendered'
+    )
+
+    const { sendEvents } = waitForApmServerCalls(0, 2)
+    const { transactions } = sendEvents
+    expect(transactions.length).toBe(2)
+
+    const pageLoadTransaction = transactions[0]
+    expect(pageLoadTransaction.type).toBe('page-load')
+    expect(pageLoadTransaction.name).toBe('/notfound')
+
+    const routeTransaction = transactions[1]
+    expect(routeTransaction.name).toBe('/lazy')
+    expect(routeTransaction.type).toBe('route-change')
+
+    expect(routeTransaction.spans.length).toBe(1)
```

Comment:
Shouldn't the request that fetches the separate JS bundle be captured from the resource timing?

Reply:
It wont, since use the pre bundled functional component, I will create a new lazy component and add it.